sessionctx/stmtctx/stmtctx.go

```diff
@@ -38,6 +39,13 @@ const (
 	WarnLevelNote = "Note"
 )
 
+var taskIDAlloc uint64
+
+// AllocateTaskID allocates a new unique ID for a statement execution
+func AllocateTaskID() uint64 {
+	return atomic.AddUint64(&taskIDAlloc, 1)
```

Comment:
There is a chance that different tidb-server use the same task ID.

Reply:
TiKV will combine both `start_ts` and `task_id` into one.

See https://github.com/tikv/tikv/pull/7521/files#diff-d674d3a094c25d9060532e4d2ff339d1R158-R178

Reply:
Got it, how about moving this calculation logic into the request sender side?

Reply:
TiDB's logic is too complicated, I'm not sure if start ts exists when task id is assigned.

Reply:
Can we use the statement idx, starts from 0 as the task id in the transaction?

Reply:
The original author also suggests to use the ID for tracing. I'm not sure if allocating as statement index will affect the purpose.